sigilrecorder.lic

```diff
@@ -0,0 +1,290 @@
+			if travel_result == true
+				sigils = find_sigils
+				record_sigil(sigils, @data) #write found sigils to file
+			else
+				#if the room was not reached successfully we assume it is unreachable and banlist it
+				DRC.message('Unable to reach target room. Adding to no-go list and moving to next room.')
+				add_to_nogo(roomid, @data) #write to nogo list in file
+			end #if
+		end #do
+
+		prune_final_list
+	end #initialize
+		
+	#returns the current season by parsing results of 'time' verb
+	def get_season
+		timestring = DRC.bput('time', /^It is currently.*/, /^The clerk says/)
+		if /^The clerk says/ =~ timestring #Annoyingly, this verb does something when near some NPCs. SIGH.
+		  DRC.message("The TIME verb does not work in this room.")
+		  add_to_nogo(Room.current.id, @data) #banlist annoying rooms where the 'time' verb works differently
+		  return false
+		end
+		/^It is currently (\w+) and it is.*$/ =~ timestring
+		return $1
+	end #get_season
+	
+	#detects the current region by comparing Room.current.image with region and city name patterns
+	def get_region
+		case Room.current.image.downcase
+		when /ilithi|Shard/i
+		  region = 'Shard'
+		when /zoluren|Crossing/i
+		  region = 'Crossing'
+		when /therengia|Riverhaven/i
+		  region = 'Riverhaven'
+		else
+		  DRC.message("Unsupported area. Sigilrecorder only supports areas where the associated map image file references Crossing, Shard, Riverhaven or their associated provinces. Current map image:#{Room.current.image}")
+		  exit
+		end #case
+		return region
+	end #get_region
+	
+	#builds a single array of all mapped sigil rooms for the current season then returns how many times roomid appears in that array
+	def check_room_data(roomid)
+		room_array = []
+		room_matches = 0
+		@sigil_list.each do |sigil|
+			@data.SigilInfo.dig(@region, sigil, @season).each do |number|
+				if number != nil
+					room_array.push(number)
+				end
+			end
+		end
+		
+		#remove nil values and then count the number of instances of roomid int the array
+		room_array.compact
+		room_matches = room_array.count(roomid)
+
+		DRC.message("Target room (#{roomid}) matches in data = #{room_matches}")
+		
+		#return the number of instances of roomid
+		return room_matches
+	end #check_room_data
+	
+	#attempts to navigate out of a room lacking important data
+	def unstuck_me(target_room)
+		#Check the exits to the room we think we're in in Mapdb. Store the exits in an array then loop through it.
+		exit_list = Map[target_room].wayto
+		exit_list.each do |rnum, dir|
+		  #flag to listen for room numbers appearing in 'peer' info
+		  Flags.add('sigil-room-detected', /Room Number: \d+/)
+		  DRC.bput("peer #{dir}", /You peer/, /You see/, /\[Usage:/)
+		  pause 0.5
+		  #if we detected a room nubmer, move in that direction and return
+		  if Flags['sigil-room-detected']
+		    DRC.retreat
+		    fput(way)
```

Comment:
❓ What variable does `way` refer to? I don't readily see it defined in this method `unstuck_me`

Reply:
This was removed and a new method of backing out of rooms that go2 doesn't recognize was implemented.